packages/eslint-plugin/src/rules/sort-interface-members.ts

```diff
@@ -0,0 +1,212 @@
+        const callSignatureDeclarations: TSCallSignatureDeclaration[] = [];
+
+        // TODO This algorithm assumes an order of TSPropertySignature > TSMethodSignature > TSIndexSignature > TSConstructSignatureDeclaration > TSCallSignatureDeclaration - it is only used to evaluate if it works alongside the member-ordering rule
+        for (let i = 0; i < members.length; i++) {
+          if (isPropertySignature(members[i])) {
+            if (
+              methodSignatures.length > 0 ||
+              indexSignatures.length > 0 ||
+              constructSignatureDeclarations.length > 0 ||
+              callSignatureDeclarations.length > 0
+            ) {
+              return context.report({
+                messageId: 'notSorted',
+                node: members[i],
+              });
+            }
+
+            propertySignatures.push(members[i] as TSPropertySignature);
+          } else if (isMethodSignature(members[i])) {
+            if (
+              indexSignatures.length > 0 ||
+              constructSignatureDeclarations.length > 0 ||
+              callSignatureDeclarations.length > 0
+            ) {
+              return context.report({
+                messageId: 'notSorted',
+                node: members[i],
+              });
+            }
+
+            methodSignatures.push(members[i] as TSMethodSignature);
+          } else if (isIndexSignature(members[i])) {
+            if (
+              constructSignatureDeclarations.length > 0 ||
+              callSignatureDeclarations.length > 0
+            ) {
+              return context.report({
+                messageId: 'notSorted',
+                node: members[i],
+              });
+            }
+
+            indexSignatures.push(members[i] as TSIndexSignature);
+          } else if (isConstructSignatureDeclaration(members[i])) {
+            if (callSignatureDeclarations.length > 0) {
+              return context.report({
+                messageId: 'notSorted',
+                node: members[i],
+              });
+            }
+
+            constructSignatureDeclarations.push(members[
+              i
+            ] as TSConstructSignatureDeclaration);
+          } else if (isCallSignatureDeclaration(members[i])) {
+            callSignatureDeclarations.push(members[
+              i
+            ] as TSCallSignatureDeclaration);
+          }
+        }
+
+        for (let i = 0; i < propertySignatures.length - 1; i++) {
+          const currentItem = <Identifier>propertySignatures[i].key;
+          const nextItem = <Identifier>propertySignatures[i + 1].key;
+
+          if (currentItem.name > nextItem.name) {
+            return context.report({
+              messageId: 'notSorted',
+              node: currentItem,
+            });
+          }
+        }
+
+        for (let i = 0; i < methodSignatures.length - 1; i++) {
+          const currentItem = <Identifier>methodSignatures[i].key;
```

Comment:
two things:
1) you shouldn't mix your assertion types within the same file.
2) we don't use the angle bracket assertions in this codebase.